### PR TITLE
IF-ELSE statement

### PR DESCRIPTION
**Pull request by avbelov23**
Adding IF-ELSE statement in grammar and stricter definition CASE statement and IF function

### CODE REVIEW

File: lib/sqli/sqli_parser.y [hunk 1]
```diff
@@ -236,6 +249,19 @@ expr:   func
             sqli_store_data(ctx, &$operator);
         }
         | expr post_exprs
+        | TOK_IF[tk] '('[u1] expr ','[u2] expr ')'[u3] {
```
Comment (sorc1) [inline lib/sqli/sqli_parser.y]: Some of the commit rules (and of the other commits) are duplicated. The original `expr` and `op_expr` rules also have duplicated rules. So, I've refactored that, see master branch. Please rewrite the commits to also remove code duplication.

**Comment by avbelov23** [comment]
DONE

File: lib/sqli/sqli_parser.y [hunk 2]
```diff
@@ -207,7 +233,16 @@ op_expr:  expr_common
             sqli_store_data(ctx, &$tk);
             YYUSE($u1);
         }
-        | op_expr post_exprs
+        | op_expr in_boolean_mode_exprs
```
Comment (sorc1) [inline lib/sqli/sqli_parser.y]: It is unclear why did you change that.

Comment (avbelov23) [reply lib/sqli/sqli_parser.y]: TOK_END (from `post_exprs`) is now used in if-else `expr`

**Comment by sorc1** [comment]
The changes are now in master branch. Thanks!
